### PR TITLE
[#3] Introduce standardized errors

### LINKED ISSUE #3: Nail error handling
Figure out the error model used while validating an extension.
Could be inspired by https://github.com/cognitect-labs/anomalies and https://github.com/dawcs/anomalies-tools

### PR DESCRIPTION
**Pull request by jeluard**
fixes #3

Reader errors are now standardized for simpler manipulation.

### CODE REVIEW

File: src/pluto/reader/errors.cljc [hunk 1]
```diff
@@ -3,18 +3,24 @@
...
+                ::unknown-reference
+                ::unknown-component
+                ::unsupported-test-type})
+
+(s/def ::value any?)
+
 (s/def ::message string?)
-(s/def ::error (s/keys :req [::type ::value]
+
+(s/def ::error (s/keys :req [::category ::value]
```
Comment (flexsurfer) [inline src/pluto/reader/errors.cljc]: `::category ` - > `::type`  ?

File: src/pluto/reader/errors.cljc [hunk 2]
```diff
@@ -3,18 +3,24 @@
   Inspired by https://github.com/cognitect-labs/anomalies and https://github.com/dawcs/anomalies-tools"
   (:require [clojure.spec.alpha :as s]))
```
Comment (flexsurfer) [inline src/pluto/reader/errors.cljc]: `:as spec` ?

File: src/pluto/reader/errors.cljc (hunk 1 above)
Comment (jeluard) [reply src/pluto/reader/errors.cljc]: Good catch!

File: src/pluto/reader/errors.cljc (hunk 2 above)
Comment (jeluard) [reply src/pluto/reader/errors.cljc]: :+1:

File: src/pluto/reader.cljc [hunk 3]
```diff
@@ -40,34 +38,36 @@
...
-           nil))}
-      (when-let [v @errors]
-        {:errors v}))))
-
-(def valid-extension-keys #{:meta})
+  (try
+    {:data
+     #?(:clj   (binding [reader/*read-eval* false]
+                 (reader/read-string {} s))
+         :cljs (reader/read-string {} s))}
```
[the remark this answers is not in the document]
Comment (jeluard) [reply src/pluto/reader.cljc]: :+1:

**Comment by jeluard** [comment]
@flexsurfer Just pushed a new commit to address your comments

**Pull request merged**